### PR TITLE
roc-toolkit: add sox support

### PR DESCRIPTION
**Pull request by chaoticryptidz**
###### Motivation for this change
adds support to roc-send to send from file sources other than 

###### Things done

replaced --disable-sox with a option to enable sox

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by chaoticryptidz** [comment]
for whoever merges this: please squash commits

**Comment by winterqt** [comment]
Is it possible for you to fix the merge conflicts (due to storage constraints), or would it be easier for you if I rebase myself and force push?

**Comment by ribosomerocker** [comment]
Any progress on this, @chaoticryptidz? it seems this was forgotten for a bit.

**Comment by chaoticryptidz** [comment]
has been ready to merge for a while nowOn 30 Dec 2022, at 12:17 pm, ribosomerocker ***@***.***> wrote:﻿
Any progress on this, @chaoticryptidz? it seems this was forgotten for a bit.

**Comment by ribosomerocker** [comment]
Nice! thank you for the quick response. We should cc @winterqt to review this then.

File: pkgs/development/libraries/audio/roc-toolkit/default.nix
Comment (wegank) [inline pkgs/development/libraries/audio/roc-toolkit/default.nix:43]: While we're at this:

```suggestion
  ]
  ++ lib.optional libunwindSupport libunwind
  ++ lib.optional openfecSupport openfec
  ++ lib.optional pulseaudioSupport libpulseaudio
  ++ lib.optional soxSupport sox;
```

Comment (winterqt) [reply pkgs/development/libraries/audio/roc-toolkit/default.nix:43]: That's why I didn't merge this when I last looked at it, but was going to add it when it was brought up again a few days ago.

Anyways, I don't think it's worth breaking compatibility for options that don't really affect us at all, providing more customization is always good.

I'll push that change as soon as I can and land this.

**Comment by winterqt** [comment]
> adds support to roc-send to send from file sources other than

Other than...? I'm afraid I can't easily tell what sources are supported without Sox. (I want to add a proper description to this.)

**Comment by ribosomerocker** [comment]
Nice! I think that's all @winterqt might need for the description

**Comment by winterqt** [comment]
Sorry for the delay. I believe this should be good to go now.

**Comment by winterqt** [comment]
@chaoticryptidz if you could give it a final OK that'd be great, thank you!

Also: if any of you are interested in becoming a maintainer for this package, that would be great, too.

**Comment by chaoticryptidz** [comment]
lgtm, willing to be maintainer for occasional version bumps, i have another pr open to add me to maintainers.nix so merge bothOn 10 Jan 2023, at 3:18 am, Winter ***@***.***> wrote:﻿
@chaoticryptidz if you could give it a final OK that'd be great, thank you!
Also: if any of you are interested in becoming a maintainer for this package, that would be great, too.

**Review by wegank — APPROVED** [review]

**Comment by winterqt** [comment]
Oh, no, this goes to staging...

**Comment by winterqt** [comment]
...there we go. Waiting on OfBorg.

**Pull request merged**
